frontend/controller/actions/identity-kv.js

```diff
@@ -1,10 +1,29 @@
 'use strict'
 import sbp from '@sbp/sbp'
 import { KV_KEYS, KV_LOAD_STATUS } from '~/frontend/utils/constants.js'
-import { KV_QUEUE, NEW_PREFERENCES, NEW_UNREAD_MESSAGES, ONLINE, NEW_KV_LOAD_STATUS } from '~/frontend/utils/events.js'
+import { debounce, difference, intersection, union } from 'turtledash'
+import { KV_QUEUE, NAMESPACE_REGISTRATION, NEW_PREFERENCES, NEW_UNREAD_MESSAGES, ONLINE, NEW_KV_LOAD_STATUS } from '~/frontend/utils/events.js'
 import { isExpired } from '@model/notifications/utils.js'
 
 const initNotificationStatus = (data = {}) => ({ ...data, read: false })
+// Name discrepancies between the KV store and `namespaceLookups` may occur
+// due to being unsubcribed from an identity contract (e.g., someone has left
+// a group) or due to the username being deleted. This function attempts to
+// determine which case it is, and determine all of the names that are currently
+// valid.
+const checkAndAugmentNames = async (currentNames: string[]) => {
+  const ourNames = Object.keys(sbp('state/vuex/state').namespaceLookups || {})
+  const unconflictedNames = intersection(currentNames, ourNames)
+  const recheckedNames = await Promise.all(difference(union(currentNames, ourNames), unconflictedNames).map(async (name) => {
+    const value = await sbp('namespace/lookup', name, { skipCache: true })
+
+    return value ? name : null
+  })).then((names) => {
+    return names.filter((v) => !!v)
+  })
+
+  return union(unconflictedNames, recheckedNames)
+}
```

Comment:
Opus has a concern:

Each lookup makes an HTTP request to the server. If there are many conflicting names, this could result in many concurrent requests. Consider:

```javascript
// Batch the lookups to avoid too many concurrent requests
const BATCH_SIZE = 10
const namesToCheck = difference(union(currentNames, ourNames), unconflictedNames)
const recheckedNames = []

for (let i = 0; i < namesToCheck.length; i += BATCH_SIZE) {
  const batch = namesToCheck.slice(i, i + BATCH_SIZE)
  const results = await Promise.all(batch.map(async (name) => {
    const value = await sbp('namespace/lookup', name, { skipCache: true })
    return value ? name : null
  }))
  recheckedNames.push(...results.filter(v => !!v))
}
```

Reply:
That's a valid concern, but not so sure about the batching. At a first glance, it seems like a good idea.